File: study/lua/src/app/data/TowerSurpriseData.lua
local BaseData = require("app.data.BaseData")
local TowerSurpriseData = class("TowerSurpriseData", BaseData)

local schema = {}
schema["surprise_id"] 			        = {"number", 0}
schema["win"] 			                = {"boolean", false}
TowerSurpriseData.schema = schema

function TowerSurpriseData:ctor(properties)
    TowerSurpriseData.super.ctor(self, properties)
end

function TowerSurpriseData:clear()

end

function TowerSurpriseData:reset()	
end

function TowerSurpriseData:updateData(data)

end

return TowerSurpriseData

File: study/ts/script/data/TowerSurpriseData.ts
import { BaseData } from "./BaseData";

export interface TowerSurpriseData {
    getSurprise_id(): number
    setSurprise_id(value: number): void
    getLastSurprise_id(): number
    isWin(): boolean
    setWin(value: boolean): void
    isLastWin(): boolean
}
let schema = {};
schema['surprise_id'] = [
    'number',
    0
];
schema['win'] = [
    'boolean',
    false
];
export class TowerSurpriseData extends BaseData {
    public static schema = schema;

    constructor(properties?) {
        super(properties);
    }
    public clear() {
    }
    public reset() {
    }
    public updateData(data) {
    }
}
